app/init2.php

```diff
@@ -0,0 +1,1371 @@
+            }
+        }
+
+        // Get fallback session from old clients (no SameSite support) or clients who block 3rd-party cookies
+        if ($response) {
+            $response->addHeader('X-Debug-Fallback', 'false');
+        }
+
+        if (empty($session['id']) && empty($session['secret'])) {
+            if ($response) {
+                $response->addHeader('X-Debug-Fallback', 'true');
+            }
+            $fallback = $request->getHeader('x-fallback-cookies', '');
+            $fallback = \json_decode($fallback, true);
+            $session = Auth::decodeSession(((isset($fallback[$authentication->getCookieName()])) ? $fallback[$authentication->getCookieName()] : ''));
+        }
+
+        $authentication->setUnique($session['id'] ?? '');
+        $authentication->setSecret($session['secret'] ?? '');
+
+        if (APP_MODE_ADMIN !== $mode) {
+            if ($project->isEmpty()) {
+                $user = new Document([]);
+            } else {
+                if ($project->getId() === 'console') {
+                    $user = $dbForConsole->getDocument('users', $authentication->getUnique());
+                } else {
+                    $user = $dbForProject->getDocument('users', $authentication->getUnique());
+                }
+            }
+        } else {
+            $user = $dbForConsole->getDocument('users', $authentication->getUnique());
+        }
+
+        if (
+            $user->isEmpty() // Check a document has been found in the DB
+            || !Auth::sessionVerify($user->getAttribute('sessions', []), $authentication->getSecret())
+        ) { // Validate user has valid login token
+            $user = new Document([]);
+        }
+
+        if (APP_MODE_ADMIN === $mode) {
+            if ($user->find('teamId', $project->getAttribute('teamId'), 'memberships')) {
+                $authorization->setDefaultStatus(false);  // Cancel security segmentation for admin users.
+            } else {
+                $user = new Document([]);
+            }
+        }
+
+        $authJWT = $request->getHeader('x-appwrite-jwt', '');
+
+        if (!empty($authJWT) && !$project->isEmpty()) { // JWT authentication
+            $jwt = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', 3600, 0);
+            try {
+                $payload = $jwt->decode($authJWT);
+            } catch (JWTException $error) {
+                $request->removeHeader('x-appwrite-jwt');
+                throw new Exception(Exception::USER_JWT_INVALID, 'Failed to verify JWT. ' . $error->getMessage());
+            }
+
+            $jwtUserId = $payload['userId'] ?? '';
+            $jwtSessionId = $payload['sessionId'] ?? '';
+
+            if ($jwtUserId && $jwtSessionId) {
+                $user = $dbForProject->getDocument('users', $jwtUserId);
+            }
+
+            if (empty($user->find('$id', $jwtSessionId, 'sessions'))) { // Match JWT to active token
+                $user = new Document([]);
+            }
```

Comment:
There's a long-standing [problem](https://github.com/appwrite/appwrite/issues/8174) where calling `account.getSession('current')` doesn't work with a JWT because we look up a session using the session secret even though the JWT has the session id. 

Can we do what [this](https://github.com/appwrite/appwrite/pull/8313) PR does to fix the problem by adding sessionId to `Authentication` and keeping track of it?